### DISCUSSION TITLE
Which font is the fallback font ?

### DISCUSSION THREAD

**Discussion by aj3423** [neovide/neovide#1952] [Q&A] [answered]
My terminal font is `Fira Code`, my terminal Neovim renders Chinese character correctly, But Neovide renders it as square. I understand there's an option `guifont`, I set it to `Fira Code` but it doesn't work, I guess nvim uses some fallback font.

It it possible to find out which Chinese font is used by terminal nvim  so I can `set guifont` with it? Or which font is the default fallback font for Chinese so I can install it?

**Comment by fredizzimo** [comment, marked as answer]
Neovide should use the platform font callback through the Skia library, but maybe it's not always working, which OS are you using?

To know which font the terminal is using is terminal dependent, so you need to check the documentation for the terminal you are using. Perhaps you have even configured a fallback yourself at some point by forgot about it.

Anyway, you should be able to use almost any monospace font, just search for Chinese mono font in Google and a lot of results will turn up. In Neovide  `guifont` you then specify both FiraCode  as the primary font and that other Chinese font as a fallback, with a comma between them. For more details see https://neovide.dev/configuration.html#font. Run `set guifont=*` to list all the font names that Neovide will accept, and make sure you use exactly those names.

**Comment by aj3423** [comment]
@fredizzimo Thanks, the `set guifont=*` is really useful, this is what I set now and it works: `vim.opt.guifont  = 'Fira Code,Noto Sans Mono CJK SC:h12'`
